### PR TITLE
Add ptr CI run

### PR DESCRIPTION
**Pull request by cooperlees**
- Since ptr was invented for terragraph we might as well keep running it
  to test the smalls bits of python left
- Run in 3.10
  - I tried python3.7 but it didn't work and I didn't debug why
- Edit `.ptrconfig` to be external friendly
  - Add some deps needed for tests to pass
- Skip and even disable tests that don't work without a yocto build or
  are broken - Would love fixes ...
- black format some files with newer black
  - Mostly docstrings ...

## Test Plan:

- Run ptr locally
  - `python3.10 -m venv /tmp/tg --upgrade-deps`
  - `/tmp/tg/bin/pip install ptr`
  - `/tmp/tg/bin/ptr -k`
```
-- Summary (total time 1s):

✅ PASS: 2
❌ FAIL: 0
⌛ TIMEOUT: 0
🔒 DISABLED: 4
💩 TOTAL: 2

-- 2 / 6 (33%) `setup.py`'s have `ptr` tests running
```

### CODE REVIEW

File: .github/workflows/python_ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: python_ci
+
+on: [push, pull_request]
```
Comment (elludraon) [inline .github/workflows/python_ci.yml:3]: Can you delete "pull_request" here? We're only running workflows on push for now, will consider pull_request later due to security concerns.

**Review by elludraon — COMMENTED** [review]
Awesome, thanks for fixing this Cooper. One comment in the workflow file.

We can merge this as a separate workflow for now but can consider adding it into the e2e-image workflow after that's finished (see #14) so that we can re-enable some of these tests which require openr.

File: .github/workflows/python_ci.yml (hunk 1 above)
Comment (cooperlees) [reply .github/workflows/python_ci.yml:3]: I can. But admins, as you just saw, have to approve first time contributors CI to run to negate this worry.

Comment (elludraon) [reply .github/workflows/python_ci.yml]: Thanks. Yeah, I understand the risk is low especially with first-time confirmation. But it's also easy to revert broken changes if needed (there's a button) and we may eventually move this workflow onto the self-hosted runner which we're more concerned about.

**Review by elludraon — APPROVED** [review]

**Pull request merged**